addons/binding/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/discovery/Enigma2DiscoveryParticipant.java

```diff
@@ -0,0 +1,127 @@
+ * @author Thomas Traunbauer - Initial contribution
+ */
+public class Enigma2DiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private Logger logger = LoggerFactory.getLogger(Enigma2DiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Enigma2BindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public DiscoveryResult createResult(ServiceInfo info) {
+        DiscoveryResult result = null;
+        ThingUID uid = getThingUID(info);
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>(4);
+            String label;
+            try {
+                label = info.getName();
+            } catch (Exception e) {
+                label = "unnamed enigma2 device";
+            }
+            // remove the domain from the name
+            InetAddress[] addrs = info.getInetAddresses();
+
+            if (addrs.length > 1) {
+                logger.warn("Enigma2 device {} ({}) reports multiple addresses - using the first one! {}",
+                        info.getName(), label, Arrays.toString(addrs));
+            }
+
+            Long defaultRefreshinterval = (long) 5000;
+            properties.put(Enigma2BindingConstants.DEVICE_PARAMETER_HOST, addrs[0].getHostAddress());
+            properties.put(Enigma2BindingConstants.DEVICE_PARAMETER_USER, "");
+            properties.put(Enigma2BindingConstants.DEVICE_PARAMETER_PASSWORD, "");
+            properties.put(Enigma2BindingConstants.DEVICE_PARAMETER_REFRESH, defaultRefreshinterval);
+            return DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(label).build();
+        }
+        return result;
+    }
+
+    @Override
+    public ThingUID getThingUID(ServiceInfo info) {
+        if (info != null) {
+            logger.debug("ServiceInfo: {}", info);
+            if (info.getType() != null) {
+                if (info.getType().equals(getServiceType())) {
+                    logger.trace("Discovered a Enigma2 device thing with name '{}'", info.getName());
+                    String formatedIP = getFormattedIPAddress(info);
+                    if (formatedIP != null) {
+                        return new ThingUID(Enigma2BindingConstants.THING_TYPE_DEVICE, new String(formatedIP));
+                    } else {
+                        return null;
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String getServiceType() {
+        // I know, this is not optimal, but currently i have no better idea!
+        return "_ssh._tcp.local.";
+    }
+
+    private boolean isIPValid(String ip) {
+        String content;
+        try {
+            content = HttpUtil.executeUrl("GET", "http://" + ip + "/web/about", 5000);
+        } catch (IOException e) {
+            return false;
+        }
+        if ((content != null) && (content.contains("e2enigmaversion"))) {
```

Comment:
Maybe return:

```java
return content != null && content.contains("e2enigmaversion"); 
```
Instead of return true and return false in the other case.